packages/react/src/components/createInlineOverlayComponent.tsx

```diff
@@ -99,7 +114,36 @@ export const createInlineOverlayComponent = <PropType, ElementType>(
 
     componentDidUpdate(prevProps: IonicReactInternalProps<PropType>) {
       const node = this.ref.current! as HTMLElement;
-      attachProps(node, this.props, prevProps);
+
+      const newProps = {
+        ...this.props,
+        delegate: this.delegate,
+      };
+
+      attachProps(
```

Comment:
Why do we need to attach the props here? Shouldn't the render function pass the updated props via `newProps`?